test/e2e/authorization.go

```diff
@@ -0,0 +1,572 @@
+			g.BeforeEach(func() {
+				tc.data.users = []string{"system:serviceaccount:api-infrastructure:api-monitoring-controller"}
+			})
+			g.When("the namespace is kube-system", func() {
+				g.BeforeEach(func() {
+					tc.data.namespaces = []string{"kube-system"}
+				})
+				g.It("should allow to update 'skipper-default-filters' ConfigMap", func() {
+					tc.data.resources = []string{"configmaps"}
+					tc.data.verbs = []string{"update"}
+					tc.data.names = []string{"skipper-default-filters"}
+					tc.run(context.TODO(), cs, true)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+				g.It("should deny to update any other ConfigMap", func() {
+					tc.data.resources = []string{"configmaps"}
+					tc.data.verbs = []string{"update"}
+					// Technically, this should result in access undecided because we allow
+					// access to 'skipper-default-filters' ConfigMap only and we haven't
+					// specified a resource name in the test case.
+					// We consider access undecided cases also as denied.
+					tc.run(context.TODO(), cs, false)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+			})
+		})
+
+		g.When("the user is k8sapi_credentials-provider", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"zalando-iam:zalando:service:k8sapi_credentials-provider"}
+			})
+			g.It("should allow to get Secrets in kube-system namespace", func() {
+				tc.data.resources = []string{"secrets"}
+				tc.data.namespaces = []string{"kube-system"}
+				tc.data.verbs = []string{"get"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+		})
+
+		g.When("the user is stups_cdp-controller", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"zalando-iam:zalando:service:stups_cdp-controller"}
+			})
+			g.When("the namespace is kube-system", func() {
+				g.BeforeEach(func() {
+					tc.data.namespaces = []string{"kube-system"}
+				})
+				g.It("should deny to get Secrets", func() {
+					tc.data.resources = []string{"secrets"}
+					tc.data.verbs = []string{"get"}
+					tc.run(context.TODO(), cs, false)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+			})
+		})
+
+	})
+
+	g.Context("For administrators", func() {
+		var tc testCase
+		g.BeforeEach(func() {
+			tc.data.groups = [][]string{{"system:masters"}}
+			tc.data.users = []string{"nmalik"}
+		})
+
+		g.When("namespace is kube-system", func() {
+			g.BeforeEach(func() {
+				tc.data.namespaces = []string{"kube-system"}
+			})
+			g.When("the resource is a Secret", func() {
+				g.BeforeEach(func() {
+					tc.data.resources = []string{"secrets"}
+				})
+				g.It("should allow read access", func() {
+					tc.data.verbs = []string{"get", "list", "watch"}
+					tc.run(context.TODO(), cs, true)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
```

Comment:
Let's also add a test for write access?

Reply:
I'll add. Thanks.

Reply:
Added it here: https://github.com/zalando-incubator/kubernetes-on-aws/pull/8536/commits/04827d63f832859c141450ad58b07440dd42b616